src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -130,6 +130,45 @@ static Object* arraySpeciesCreate(ExecutionState& state, Object* originalArray,
     return Object::construct(state, C, 1, argv);
 }
 
+// http://ecma-international.org/ecma-262/10.0/#sec-flattenintoarray
+// FlattenIntoArray(target, source, sourceLen, start, depth [ , mapperFunction, thisArg ])
+static int64_t flattenIntoArray(ExecutionState& state, Value target, Value source, int64_t sourceLen, int64_t start, double depth, Value mappedValue = Value(), Value thisArg = Value())
```

Comment:
`EmptyValue` is an internal value to represent any invalid value. It would be better to explicitly set default value of optional parameters (mappedValue, thisArg) as `Value(Value::EmptyValue)` because `undefined` could be passed as a valid argument. We can check the empty value by calling the `Value::isEmpty`.